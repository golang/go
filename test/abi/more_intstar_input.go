// run

//go:build !wasm

// Copyright 2021 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

// wasm is excluded because the compiler chatter about register abi pragma ends up
// on stdout, and causes the expected output to not match.

package main

var sink int

//go:registerparams
//go:noinline
func F(a, b, c, d, e, f, g, h, i, j, k, l, m *int) {
	G(m, l, k, j, i, h, g, f, e, d, c, b, a)
	// did the pointers get properly updated?
	sink = *a + *m
}

//go:registerparams
//go:noinline
func G(a, b, c, d, e, f, g, h, i, j, k, l, m *int) {
	// Do not reference the parameters
	var scratch [1000 * 100]int
	I := *c - *e - *l // zero.
	scratch[I] = *d
	println("Got this far!")
	sink += scratch[0]
}

func main() {
	a, b, c, d, e, f, g, h, i, j, k, l, m := 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13
	F(&a, &b, &c, &d, &e, &f, &g, &h, &i, &j, &k, &l, &m)
	println("Sink =", sink-7)
}
