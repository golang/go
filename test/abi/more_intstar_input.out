Got this far!
Sink = 7
